docs/development/design/fsm.md

```diff
@@ -0,0 +1,57 @@
+# Nexd State Machine
+
+> [Issue #885](https://github.com/nexodus-io/nexodus/issues/885)
+
+## Summary
+
+State management in nexd is becoming more complicated.
+We need a mechanism to allow us to simplify that logic and not rely on crash/restart to retrigger things.
+
+## Proposal
+
+I propose that we implement a Finite State Machine inside of nexd that operates as follows:
+
+```mermaid
+---
+title: Nexd State Machine
+---
+stateDiagram-v2
+    [*] --> IDLE
+    IDLE --> AUTH: "login"
+    AUTH --> READY: "login_success"
+    AUTH --> IDLE: "login_failed"
+    READY --> PEERING: "start"
+    READY --> IDLE: "logout"
+    PEERING --> READY: "stop"
+    PEERING --> READY: "peering_error"
+    PEERING --> IDLE: "authentication_error"
+    PEERING --> IDLE: "logout"
+```
+
+nexd may be in one of four states:
+
+1. IDLE: No configuration has been provided
+2. AUTH: Performing authentication, either interactively or via token
+3. READY: Authenticated and ready to peer
+4. PEERING: Actively Peering
+
+To provide information to the user about the current state, a `nexctl nexd status` command will be added. This can inform the user of the current state and what to do next.
+
+To transition from IDLE to AUTH the "login" transition must occur.
+
+The trigger for this will be a CLI command: `nexctl nexd login`
+
+Once in the "AUTH" state, credentials will be taken from the user - or the device-auth flow will occur. In event of successful login the "login_success" transition occurs to bring nexd into the READY state. In case of "login_failed" we go back to IDLE.
+
+When in the READY state we will immediately issue a transition "start" into the PEERING state to preserve the existing behavior. The commands `nexctl nexd start` and `nexctl nexd stop` will be added to manually issue these transitions so users can choose to start/stop peering as necessary.
```

Comment:
Another option if you look at it from bringing network interfaces up and down would be: `nexctl nexd up` and `nexctl nexd down`

Reply:
Also think we should not in the design that doing a `nexctl nexd up` or `nexctl nexd start` implicitly does a `nexctl nexd login` if your not logged in yet.

Reply:
>Another option if you look at it from bringing network interfaces up and down would be: nexctl nexd up and nexctl nexd down it would also avoid confusion from the user thinking your starting or stopping nexd.

But I think It creates another level of confusion because interface up doesn't really mean that it's connected to the network (if we really associated `up/down` with interface up/down). Another option can be `nexctl nexd join`, `nexctl nexd leave`.

Reply:
Ok... sounds sound like maybe we need more states? How about: 

DISCONNECTED: No configuration has been provided, not logged in
CONNECTING: Performing authentication, either interactively or via token
UNREGISTERED: Authenticated but devices is not registered with the control plane
REGISTERED: Authenticated and registered with the control plane
DOWN: Registered but the wg interface is not up yet
UP: wg interface is up and is peering.

Reply:
States have been renamed thanks.